Add hotsos documentation (#652)
canonical/hotsos: doc/source/contrib/scenarios.rst

Overview
=========

Scenarios provide a way to define analysis in terms of :ref:`checks` and
:ref:`conclusions` where the latter are derived from decisions based
on the outcome of one or more checks.

Scenarios run automatically and are written in YAML. See existing

> definitions for examples. They grouped by plugin at the top level

Reviewer: ```suggestion
definitions for examples. They are grouped by plugin at the top level
```

and groupings beneath are purely logical/organisational. A file may contain

> one or more scenario where a scenario must define its own set of

Reviewer: ```suggestion
one or more scenarios where a scenario must define its own set of
```

:ref:`checks` and :ref:`conclusions`. Typically they are

> implemented as a one scenario per file.

Reviewer: ```suggestion
implemented as one scenario per file.
```

> Checks are implemented independently of each other and the results saved for

Reviewer: ```suggestion
Checks are implemented independently of each other and the results are saved for
```

subsequent :ref:`decision` when forming :ref:`conclusions`.
Conclusions are defined as a decision based on the outcome of one or more
:ref:`checks` along with information such as the issue and message to
raise if a conclusion is matched. Conclusions can be given
:ref:`priority` so that one can be selected in the event of multiple
positives. This is helpful for defining fallback conclusions.

The following propeties are mandatory when writing a scenario:

  * :ref:`checks`
  * :ref:`conclusions`

> See property reference for details on how to configure these properties.

Reviewer: ```suggestion
See :ref:`property_ref<property reference>` for details on how to configure these properties.
```

Example Scenario
================

> Lets say for example that we have a systemd service called *neverfail* that

Reviewer: ```suggestion
Suppose for example that we have a systemd service called *neverfail* that
```
